test/src/matrix3.spec.js

```diff
@@ -0,0 +1,283 @@
+
+  const INPUT = INDICES_MATRIX;
+  const RESULT = TRANSPOSED_INDICES_MATRIX;
+
+  let m = new Matrix3().setRowMajor(...INPUT);
+
+  tapeEquals(t, m, RESULT, 'setRowMajor gave the right result');
+
+  m = new Matrix3().setRowMajor(1, 2, 3, 4, 5, 6);
+
+  tapeEquals(
+    t,
+    m,
+    [1, 4, 0, 2, 5, 0, 3, 6, 1],
+    'setRowMajor with missing params gave the right result'
+  );
+
+  m = new Matrix3().setRowMajor();
+
+  tapeEquals(t, m, IDENTITY_MATRIX, 'setRowMajor with no params gave the right result');
+
+  t.end();
+});
+
+test('Matrix3#setColumnMajor', t => {
+  t.equals(typeof Matrix3.prototype.setColumnMajor, 'function');
+
+  const INPUT = INDICES_MATRIX;
+  const RESULT = INDICES_MATRIX;
+
+  let m = new Matrix3().setColumnMajor(...INPUT);
+
+  tapeEquals(t, m, RESULT, 'set gave the right result');
+
+  m = new Matrix3().setColumnMajor(1, 2, 3, 4, 5, 6);
+
+  tapeEquals(
+    t,
+    m,
+    [1, 2, 3, 4, 5, 6, 0, 0, 1],
+    'setColumnMajor with missing params gave the right result'
+  );
+
+  m = new Matrix3().setColumnMajor();
+
+  tapeEquals(t, m, IDENTITY_MATRIX, 'setColumnMajor with no params gave the right result');
+
+  t.end();
+});
+
+test('Matrix3#set', t => {
+  t.equals(typeof Matrix3.prototype.set, 'function');
+
+  const INPUT = INDICES_MATRIX;
+  const RESULT = INDICES_MATRIX;
+
+  const m = new Matrix3().set(...INPUT);
+
+  tapeEquals(t, m, RESULT, 'set gave the right result');
+
+  t.end();
+});
+
+test('Matrix3#getElement and setElement', t => {
+  t.equals(typeof Matrix3.prototype.setElement, 'function');
+  t.equals(typeof Matrix3.prototype.getElement, 'function');
+
+  const INPUT = INDICES_MATRIX;
+
+  const m = new Matrix3().set(...INPUT);
+
+  const VALUE = 10;
+
+  m.setElement(2, 1, VALUE);
+  let result = m.getElement(2, 1);
+  tapeEquals(t, result, VALUE, 'getElement gave the right result');
+
+  m.setElement(2, 1, VALUE, true);
+  result = m.getElement(2, 1, true);
+  tapeEquals(t, result, VALUE, 'getElement gave the right result');
+
+  t.end();
+});
+
+test('Matrix3#determinant', t => {
+  const RESULT = 5;
+
+  t.equals(typeof Matrix3.prototype.determinant, 'function');
+  const m = new Matrix3().set(1, 2, 3, 0, 1, 5, 5, 6, 0);
+  const result = m.determinant();
+
+  tapeEquals(t, result, RESULT, 'determinant gave the right result');
+  t.end();
+});
+
+test('Matrix3#identity (identity matrix)', t => {
+  t.equals(typeof Matrix3.prototype.identity, 'function');
+  const m = new Matrix3();
+  m.identity();
+
+  const RESULT = IDENTITY_MATRIX;
+
+  tapeEquals(t, m, RESULT);
+  t.end();
+});
+
+test('Matrix3#copy', t => {
+  t.equals(typeof Matrix3.prototype.copy, 'function');
+
+  const INPUT = INDICES_MATRIX;
+  const RESULT = INDICES_MATRIX;
+
+  const m = new Matrix3().copy(INPUT);
+
+  tapeEquals(t, m, RESULT, 'copy gave the right result');
+
+  t.end();
+});
+
+test('Matrix3#fromQuaternion', t => {
+  t.equals(typeof Matrix3.prototype.fromQuaternion, 'function');
+
+  const RESULT = [
```

Comment:
Where is the expected result from? Add a comment on how future maintainers can validate its correctness.